### PR TITLE
Maybe default the selected state as the first one when it's "*"

### PR DESCRIPTION
**Pull request by sc0ttkclark**
### Changes proposed in this Pull Request:

When WC is provided only the country for the option `woocommerce_default_country` then it will fail to select a default state within the listed states.

Try this by manually setting `woocommerce_default_country` to "US" and you will see it selects the first country in the list instead of the first state within that country.

This only happens for countries that have states like US and CA for instance.

If the default state is intended to be "*" which is getting passed to this function, then that logic should pass along to countries with states too so that it will always default to the first one. Otherwise, the country never gets selected and that flow is broken.

### How to test the changes in this Pull Request:

1. Create a WC store (but don't complete onboarding)
2. Manually set option: `wp option update woocommerce_default_country US` (use US or CA or any other country that has states that are selectable)
3. Go to the WooCommerce > Settings page
4. Look at the "Country / State" option, it should be now default to the first state for that country (previously it would set as "Afghanistan")

### CODE REVIEW

File: plugins/woocommerce/includes/class-wc-countries.php [hunk 1]
```diff
@@ -474,6 +474,11 @@ public function country_dropdown_options( $selected_country = '', $selected_stat
 			foreach ( $this->countries as $key => $value ) {
 				$states = $this->get_states( $key );
 				if ( $states ) {
+					// Maybe default the selected state as the first one.
+					if ( '*' === $selected_state ) {
```
Comment (rrennick) [inline plugins/woocommerce/includes/class-wc-countries.php]: If you test with `wp option update woocommerce_default_country CA`, it defaults to `Afganistan`.

```suggestion
					if ( $selected_country === $key && '*' === $selected_state ) {
```

File: plugins/woocommerce/includes/class-wc-countries.php [hunk 2]
```diff
@@ -474,6 +474,11 @@ public function country_dropdown_options( $selected_country = '', $selected_stat
...
+						$selected_state = key( $states );
```
Comment (rrennick) [inline plugins/woocommerce/includes/class-wc-countries.php]: We only want to select a default state in countries that have states.

```suggestion
						$selected_state = key( $states ) ?? '*';
```

**Review by rrennick — CHANGES_REQUESTED** [review]
@sc0ttkclark Thanks for submitting this. The suggestions below should ensure this works correctly for all countries.

**Comment by sc0ttkclark** [comment]
Sounds good, I'll get those implemented

**Review by rrennick — APPROVED** [review]
@sc0ttkclark Thanks for a great convenience fix for merchants setting up a new store.

**Pull request merged**